### PR TITLE
Add configs for migrating monitoring-setup containers to new instance

### PR DESCRIPTION
**Pull request by burnerlee**
Add configs to set up Prometheus, grafana, push_gateway and alert-manager on a separate EC2 instance.
Deliverables:
- Add nginx container to setup reverse proxy for monitoring instance
- Add auto deployment commands for monitoring instance
- Change prometheus scrape targets accordingly

### CODE REVIEW

**Comment by burnerlee** [comment]
Need to update the auto-deployment script once the environment variable and filenames are finalised

File: docker-compose-staging.yml [hunk 1]
```diff
@@ -143,3 +146,18 @@ services:
       - '--web.route-prefix=/push_gateway'
     ports:
       - '9091:9091'
+  
+  monitoring-nginx:
```
Comment (Ram81) [inline docker-compose-staging.yml:150]: Rename it to `nginx-ingress`

File: docker-compose-staging.yml [hunk 2]
```diff
@@ -143,3 +146,18 @@ services:
...
+    image: ${AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-staging-monitoring-nginx:${COMMIT_ID}
+    build:
+      context: ./
+      dockerfile: docker/prod/monitoring-nginx/Dockerfile
```
Comment (Ram81) [inline docker-compose-staging.yml:154]: same here

File: monitoring/prometheus/prometheus_staging.yml [hunk 3]
```diff
@@ -21,4 +22,4 @@ scrape_configs:
   - job_name: 'push_gateway'
     metrics_path: '/push_gateway/metrics'
     static_configs:
-      - targets: ['staging.eval.ai']
+      - targets: ['push_gateway:9091']
```
Comment (Ram81) [inline monitoring/prometheus/prometheus_staging.yml:25]: This is not correct. If we are moving to separate nginx we should avoid referencing by container names. This should be a valid endpoint dns

File: scripts/deployment/deploy.sh [hunk 4]
```diff
@@ -47,7 +49,18 @@ case $opt in
 					aws s3 cp s3://cloudcv-secrets/evalai/${env}/docker_${env}.env ./docker/prod/docker_${env}.env
 					docker-compose -f docker-compose-${env}.yml rm -s -v -f
 					docker-compose -f docker-compose-${env}.yml pull
-					docker-compose -f docker-compose-${env}.yml up -d --force-recreate --remove-orphans statsd-exporter django nodejs nodejs_v2 celery node_exporter push_gateway
+					docker-compose -f docker-compose-${env}.yml up -d --force-recreate --remove-orphans statsd-exporter django nodejs nodejs_v2 celery node_exporter
+				ENDSSH2
+                ssh ubuntu@${MONITORING_INSTANCE} -o StrictHostKeyChecking=no AWS_ACCOUNT_ID=${AWS_ACCOUNT_ID} COMMIT_ID=${COMMIT_ID} env=${env} 'bash -s' <<-'ENDSSH2'
```
Comment (Ram81) [inline scripts/deployment/deploy.sh:54]: Fix indentation

File: docker/prod/monitoring-nginx/nginx_staging.conf [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+
+  access_log /var/log/nginx/access.log;
+  error_log /var/log/nginx/error.log;
+
+  location /prometheus {
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header Host $host;
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_pass http://prometheus;
```
Comment (Ram81) [inline docker/prod/monitoring-nginx/nginx_staging.conf:41]: As we are adding ssl certs these should be `https` endpoints

File: docker/prod/monitoring-nginx/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,3 @@
+FROM nginx:1.13-alpine
+COPY docker/prod/monitoring-nginx/nginx_${MONITORING_ENV}.conf /etc/nginx/conf.d/default.conf
```
Comment (Ram81) [inline docker/prod/monitoring-nginx/Dockerfile:2]: rename the `monitoring-nginx` folder to `nginx-ingress`

File: scripts/deployment/deploy.sh [hunk 7]
```diff
@@ -47,7 +49,18 @@ case $opt in
...
+					source venv/bin/activate
```
Comment (Ram81) [inline scripts/deployment/deploy.sh:55]: Why do you need to activate virtualenv here?

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by Ram81** [comment]
@burnerlee please test the nginx ingress changes locally before we merge it to staging.

**Comment by burnerlee** [comment]
> @burnerlee please test the nginx ingress changes locally before we merge it to staging. 

@Ram81 yeah these changes have been tested on local and then I removed the changes for dev environment before pushing the code. I'll test the changes again after making the above mentioned changes

File: scripts/deployment/deploy.sh (hunk 4 above)
Comment (burnerlee) [reply scripts/deployment/deploy.sh:54]: image
@Ram81 this looks like this in VS-code, I think the view here seems distored.... does the view of VScode seems fine?

Comment (Ram81) [reply scripts/deployment/deploy.sh:54]: Your vscode settings might not be correct. It should show correct indentation in the PR. Please make sure you are using spaces in vscode to make sure it is consistent

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Review by Ram81 — CHANGES_REQUESTED** [review]
@burnerlee please rename `beta-staging.eval.ai` to `monitoring-staging.eval.ai`

**Review by Ram81 — CHANGES_REQUESTED** [review]
@burnerlee why are there alertmanager data and nflog files part of this PR. Please remove them

**Review by Ram81 — APPROVED** [review]
LGTM

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**